src/main/java/mate/academy/springboot/datajpa/service/mapper/ProductDtoMapper.java

```diff
@@ -0,0 +1,25 @@
+package mate.academy.springboot.datajpa.service.mapper;
+
+import mate.academy.springboot.datajpa.model.Product;
+import mate.academy.springboot.datajpa.model.dto.request.ProductRequestDto;
+import mate.academy.springboot.datajpa.model.dto.response.ProductResponseDto;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Mapper(componentModel = "spring")
+public abstract class ProductDtoMapper implements DtoMapper<Product,
+        ProductResponseDto,
+        ProductRequestDto> {
+    @Autowired
+    protected CategoryService categoryService;

```

Comment:
Let's not use services in mappers, we do not need all those calls to fetch the data that we do not use, ID is enough to build a relation without fetching all other entity data + we had a point about that in one of the previous tasks checklist